File: Calculator.pas
program Kalkulator;

uses crt;

var
  a, b, hasil: Real;
  operasi: Char;

begin
  clrscr;
  writeln('=== Kalkulator Sederhana ===');
  write('Masukkan angka pertama: ');
  readln(a);
  write('Masukkan operasi (+, -, *, /): ');
  readln(operasi);
  write('Masukkan angka kedua: ');
  readln(b);

  case operasi of
    '+': hasil := a + b;
    '-': hasil := a - b;
    '*': hasil := a * b;
    '/': 
      if b <> 0 then
        hasil := a / b
      else
      begin
        writeln('Error: Pembagian dengan nol!');
        halt;
      end;
  else
    begin
      writeln('Operasi tidak dikenal.');
      halt;
    end;
  end;

  writeln('Hasil: ', hasil:0:2);
  readln;
end.

File: Calculator.f90
program calculator
  implicit none
  real :: a, b, result
  character(1) :: op

  print *, '=== Kalkulator Sederhana ==='
  print *, 'Masukkan angka pertama:'
  read *, a
  print *, 'Masukkan operator (+, -, *, /):'
  read *, op
  print *, 'Masukkan angka kedua:'
  read *, b

  select case(op)
    case('+')
      result = a + b
    case('-')
      result = a - b
    case('*')
      result = a * b
    case('/')
      if (b /= 0.0) then
        result = a / b
      else
        print *, 'Error: Pembagian dengan nol!'
        stop
      end if
    case default
      print *, 'Operator tidak dikenali.'
      stop
  end select

  print *, 'Hasil: ', result

end program calculator
